File: frontend/src/api/equipment.ts
// ============================================================
// Equipment 设备仪器 API（B13）
// 网关 /api/stores → store-service:8085，类路径 /api/stores。
// 读接口金额单位「元」（purchaseAmount/depreciated/cost，后端已换算）；
// 写接口金额单位「分」（purchaseAmountFen/costFen），前端由「元」换算。
// 日期均为 yyyy-MM-dd（后端 LocalDate）；前端传 ISO 字符串，后端截前 10 位解析。
// ============================================================
import client from './client'

/** 校准/维保/维修记录行（嵌在设备内）。 */
export interface MaintenanceRecordDTO {
  id: number
  type: 'CALIBRATION' | 'MAINTENANCE' | 'REPAIR'
  at: string | null
  by: string
  vendor: string | null
  summary: string
  nextAt: string | null
  cost: number // 元
}

/** 设备台账行（GET /stores/equipments）；金额单位「元」。 */
export interface EquipmentDTO {
  id: number
  storeCode: string
  assetNo: string
  name: string
  brand: string | null
  model: string | null
  category: 'LASER' | 'RF' | 'ULTRASOUND' | 'INJECTION' | 'MONITOR' | 'OTHER'
  location: string
  status: 'NORMAL' | 'CALIBRATING' | 'REPAIRING' | 'DISABLED'
  purchasedAt: string | null
  purchaseAmount: number // 元
  lifespanYears: number
  depreciated: number // 元
  nextCalibrationAt: string | null
  nextMaintenanceAt: string | null
  note: string | null
  records: MaintenanceRecordDTO[]
}

/** 设备建档入参；金额 purchaseAmountFen 单位「分」，日期为 ISO 字符串。 */
export interface CreateEquipmentCmd {
  storeCode: string
  assetNo: string
  name: string
  brand?: string
  model?: string
  category: string
  location?: string
  status?: string
  purchasedAt: string
  purchaseAmountFen: number
  lifespanYears: number
  nextCalibrationAt?: string
  nextMaintenanceAt?: string
  note?: string
}

/** 状态变更入参。 */
export interface EquipmentStatusCmd {
  storeCode: string
  status: string
  note?: string
}

/** 校准/维保/维修记录入参；费用 costFen 单位「分」，at/nextAt 为 ISO 字符串。 */
export interface EquipmentRecordCmd {
  storeCode: string
  type: string
  summary: string
  vendor?: string
  at?: string
  nextAt?: string
  costFen?: number
}

/** 设备台账（含记录嵌套）；门店/自助角色后端强制本店，集团/品牌可参 storeCode。 */
export const listEquipments = (params?: {
  storeCode?: string
  category?: string
  status?: string
  keyword?: string
}) => client.get<EquipmentDTO[]>('/stores/equipments', { params })

/** 单台设备详情（含记录）。 */
export const getEquipment = (id: number, params?: { storeCode?: string }) =>
  client.get<EquipmentDTO>(`/stores/equipments/${id}`, { params })

/** 设备建档；返回 {id, assetNo, storeCode}。 */
export const createEquipment = (cmd: CreateEquipmentCmd) =>
  client.post<{ id: number; assetNo: string; storeCode: string }>('/stores/equipments', cmd)

/** 设备状态变更；返回 {ok:true}。 */
export const setEquipmentStatus = (id: number, cmd: EquipmentStatusCmd) =>
  client.post<{ ok: boolean }>(`/stores/equipments/${id}/status`, cmd)

/** 登记校准/维保/维修记录（回写下次日期/状态/折旧）；返回 {ok:true}。 */
export const addEquipmentRecord = (id: number, cmd: EquipmentRecordCmd) =>
  client.post<{ ok: boolean }>(`/stores/equipments/${id}/records`, cmd)

File: backend/store-service/src/main/java/com/meiyun/store/equipment/Equipment.java
package com.meiyun.store.equipment;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.OffsetDateTime;

/**
 * 设备仪器台账（B13）：激光/射频/超声/注射等医美仪器主数据 + 折旧/校准维保临期。
 *
 * <p>状态四态全部真实持久化：NORMAL 正常 / CALIBRATING 校准中 / REPAIRING 维修中 / DISABLED 停用。
 * 金额一律 Long「分」落库（purchase_amount_fen / depreciated_fen），出参转「元」。
 * 日期（购置/下次校准/下次维保）按 LocalDate 落库，仅精确到天。
 */
@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "equipment", uniqueConstraints = {
        @UniqueConstraint(name = "uk_eq_store_asset", columnNames = {"store_code", "asset_no"})
})
public class Equipment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "store_code", nullable = false, length = 16)
    private String storeCode;

    /** 资产编号，店内唯一 */
    @Column(name = "asset_no", nullable = false, length = 32)
    private String assetNo;

    @Column(nullable = false, length = 64)
    private String name;

    @Column(length = 64)
    private String brand;

    @Column(length = 64)
    private String model;

    /** LASER/RF/ULTRASOUND/INJECTION/MONITOR/OTHER */
    @Column(nullable = false, length = 16)
    private String category;

    @Column(nullable = false, length = 64)
    private String location;

    /** NORMAL/CALIBRATING/REPAIRING/DISABLED */
    @Column(nullable = false, length = 16)
    private String status;

    @Column(name = "purchased_at", nullable = false)
    private LocalDate purchasedAt;

    /** 购置金额（分） */
    @Column(name = "purchase_amount_fen", nullable = false)
    private Long purchaseAmountFen;

    @Column(name = "lifespan_years", nullable = false)
    private Integer lifespanYears;

    /** 已累计折旧/维保费用（分），封顶购置金额 */
    @Column(name = "depreciated_fen", nullable = false)
    private Long depreciatedFen;

    @Column(name = "next_calibration_at")
    private LocalDate nextCalibrationAt;

    @Column(name = "next_maintenance_at")
    private LocalDate nextMaintenanceAt;

    @Column(length = 255)
    private String note;

    @Column(name = "created_by", length = 32)
    private String createdBy;

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_by", length = 32)
    private String updatedBy;

    @Column(name = "updated_at")
    private OffsetDateTime updatedAt;

    @PrePersist
    void prePersist() {
        if (createdAt == null) createdAt = OffsetDateTime.now();
        if (status == null) status = "NORMAL";
        if (depreciatedFen == null) depreciatedFen = 0L;
    }
}
